src/main/java/com/aws/greengrass/util/platforms/unix/linux/CGroupSubSystemPaths.java

```diff
@@ -0,0 +1,126 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package com.aws.greengrass.util.platforms.unix.linux;
+
+import com.aws.greengrass.lifecyclemanager.GreengrassService;
+import edu.umd.cs.findbugs.annotations.SuppressFBWarnings;
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
+@SuppressFBWarnings(value = "DMI_HARDCODED_ABSOLUTE_FILENAME",
+        justification = "CGroupSubSystemPath virtual filesystem path cannot be relative")
+public interface CGroupSubSystemPaths {
+    Path CGROUP_ROOT = Paths.get("/sys/fs/cgroup");
+    String GG_NAMESPACE = "greengrass";
+    String CGROUP_MEMORY_LIMITS = "memory.limit_in_bytes";
+    String CPU_CFS_PERIOD_US = "cpu.cfs_period_us";
+    String CPU_CFS_QUOTA_US = "cpu.cfs_quota_us";
+    String CGROUP_PROCS = "cgroup.procs";
+    String FREEZER_STATE_FILE = "freezer.state";
+    String CPU_MAX = "cpu.max";
+    String MEMORY_MAX = "memory.max";
+    String CGROUP_SUBTREE_CONTROL = "cgroup.subtree_control";
+    String CGROUP_FREEZE = "cgroup.freeze";
+    String MOUNT_PATH = "/proc/self/mounts";
+    String UNICODE_SPACE = "\\040";
+
+    default Path getRootPath() {
+        return CGROUP_ROOT;
+    }
+
+    String rootMountCmd();
+
+    default String subsystemMountCmd() {
+        return null;
+    }
+
+    Path getSubsystemRootPath();
+
+    default Path getSubsystemGGPath() {
+        return getSubsystemRootPath().resolve(GG_NAMESPACE);
+    }
+
+    default Path getSubsystemComponentPath(String componentName) {
+        return getSubsystemGGPath().resolve(componentName);
+    }
+
+    Path getComponentMemoryLimitPath(String componentName);
+
+    default Path getComponentCpuPeriodPath(String componentName) {
+        return null;
+    }
+
+    default Path getComponentCpuQuotaPath(String componentName) {
+        return null;
+    }
+
+    default Path getCgroupProcsPath(String componentName) {
+        return getSubsystemComponentPath(componentName).resolve(CGROUP_PROCS);
+    }
+
+    Path getCgroupFreezerStateFilePath(String componentName);
+
+    default Path getRootSubTreeControlPath() {
+        return null;
+    }
+
+    default Path getGGSubTreeControlPath() {
+        return null;
+    }
+
+    default Path getComponentCpuMaxPath(String componentName) {
+        return null;
+    }
+
+    default Path getCgroupFreezePath(String componentName) {
+        return null;
+    }
```

Comment:
Why are these defaults `null`? If these must be implemented, don't provide a `default`, so then the must be implemented by the subclass.

Reply:
Do not just make a change, please explain what you are thinking.

Reply:
Oh,they are not necessary any more, I will remove them from the interface.At the beginning,I want to the controller share the same method.